x/ecocredit/server/testsuite/suite.go

```diff
@@ -1038,6 +1047,86 @@ func (s *IntegrationTestSuite) TestScenario() {
 	})
 	s.Require().Nil(err)
 	s.Require().Equal(expectedSellOrderIds, createSellOrder.SellOrderIds)
+	orderId1 := createSellOrder.SellOrderIds[0]
+	orderId2 := createSellOrder.SellOrderIds[1]
+
+	// now we buy these orders
+	buyerAcc := acc5
+	expectedTotalCost := sdk.NewInt64Coin(coinPrice.Denom, 26_086_420)
+	// this is the exact amount it should cost to purchase both orders
+	s.fundAccount(buyerAcc, sdk.Coins{expectedTotalCost})
+
+	buyerAccBefore := s.getAccountInfo(buyerAcc, batchDenom, coinPrice.Denom)
+	sellerAccBefore := s.getAccountInfo(acc3, batchDenom, coinPrice.Denom)
+	_, err = s.marketServer.BuyDirect(s.ctx, &marketplace.MsgBuyDirect{
+		Buyer: buyerAcc.String(),
+		Orders: []*marketplace.MsgBuyDirect_Order{
+			{
+				SellOrderId:            orderId1,
+				Quantity:               order1Qty,
+				BidPrice:               &coinPrice,
+				DisableAutoRetire:      false,
+				RetirementJurisdiction: "US-OR",
+			},
+			{
+				SellOrderId:       orderId2,
+				Quantity:          order2Qty,
+				BidPrice:          &coinPrice,
+				DisableAutoRetire: true,
+			},
+		},
+	})
+	s.Require().NoError(err)
+	buyerAccAfter := s.getAccountInfo(buyerAcc, batchDenom, coinPrice.Denom)
+	sellerAccAfter := s.getAccountInfo(acc3, batchDenom, coinPrice.Denom)
+
+	s.assertSellerBalancesUpdated(sellerAccBefore, sellerAccAfter, order2QtyDec, order1QtyDec, expectedTotalCost)
+	s.assertBuyerBalancesUpdated(buyerAccBefore, buyerAccAfter, order2QtyDec, order1QtyDec, expectedTotalCost)
+}
+
+type accountInfo struct {
+	address                     sdk.AccAddress
+	tradable, retired, escrowed math.Dec
+	bankBalance                 sdk.Coin
+}
+
+func (s *IntegrationTestSuite) assertSellerBalancesUpdated(accBefore, accAfter accountInfo, tradable, retired math.Dec, totalCost sdk.Coin) {
+	expectedEscrowed := accBefore.escrowed // account before the order was bought. should have tradable + retired in escrow
+
+	// subtract the tradable+retired amounts from escrow
+	var err error
+	expectedEscrowed, err = expectedEscrowed.Sub(tradable)
+	s.Require().NoError(err)
+	expectedEscrowed, err = expectedEscrowed.Sub(retired)
+	s.Require().NoError(err)
+
+	s.Require().True(expectedEscrowed.Equal(accAfter.escrowed), fmt.Sprintf("expected %v, got %v", expectedEscrowed, accAfter.escrowed))
+	s.Require().True(accBefore.tradable.Equal(accAfter.tradable))
+	s.Require().True(accBefore.retired.Equal(accAfter.retired))
+
+	expectedBankBal := accBefore.bankBalance
+	expectedBankBal = expectedBankBal.Add(totalCost)
+	s.Require().True(expectedBankBal.Equal(accAfter.bankBalance))
```

Comment:
Why not use `s.Require().Equal` here?

`Equal` will check the underlying values. For example, the following fails:

```
	dec1, err := math.NewDecFromString("10")
	dec2, err := math.NewDecFromString("20")
	s.Require().Equal(dec1, dec2)
```

Reply:
Also, why include a message for some and not others? If you use `Equal`, the expected and actual values will be printed in the logs. The logs for the example above:

```
                Error:          Not equal: 
                                expected: math.Dec{dec:apd.Decimal{Form:0, Negative:false, Exponent:0, Coeff:big.Int{neg:false, abs:big.nat{0xa}}}}
                                actual  : math.Dec{dec:apd.Decimal{Form:0, Negative:false, Exponent:0, Coeff:big.Int{neg:false, abs:big.nat{0x14}}}}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -8,3 +8,3 @@
                                    abs: (big.nat) (len=1) {
                                -    (big.Word) 10
                                +    (big.Word) 20
                                    }
```

Reply:
`Equal` doesn't always work, for example switching everything to `Equal` makes the test fail because of a simple difference `abs:big.nat{}` vs `abs:big.nat(nil)` in `expectedEscrow` vs `accAfter.escrowed`. 

using the decimal's `Equal` method gives a better concrete check over the internal values of the decimal, whereas `require.Equal` doesnt. 

> why include a message for some and not others

the one thats there was leftover from debugging some math errors when i was writing the tests

Reply:
Hmm... looks like it's only a problem with escrowed. Another option would be to use `String()`, which will give you the actual and expected results rather than `expected true` or `expected false`.

Reply:
ok, switched in https://github.com/regen-network/regen-ledger/pull/1079/commits/f6cc432464dbdc87af40eaad9662d579e8e13815